packages/mantine/src/components/collection/CollectionItem.tsx

```diff
@@ -8,20 +8,23 @@ import {ActionIcon} from '../action-icon';
 import {useCollectionContext} from './CollectionContext';
 
 interface CollectionItemProps extends CollectionItemSharedProps {
+    data_index: number;
     draggable?: boolean;
     disabled: boolean;
 }
 
 interface CollectionItemSharedProps extends GroupProps {
     id: string;
+    data_index: number;
     onRemove?: React.MouseEventHandler<HTMLButtonElement>;
     removable?: boolean;
 }
 
 const RemoveButton: FunctionComponent<{
+    data_index: number;
     onClick: React.MouseEventHandler<HTMLButtonElement>;
-}> = ({onClick}) => (
-    <ActionIcon.Quaternary style={{alignSelf: 'center'}} onClick={onClick}>
+}> = ({onClick, data_index}) => (
+    <ActionIcon.Quaternary data-testid={`remove-${data_index}`} style={{alignSelf: 'center'}} onClick={onClick}>
```

Comment:
Instead of adding the test id to the remove button directly, I am wondering if instead it would not be better to simply add the test id on the CollectionItem.

This way, devs could easily do something like this in their tests which follows more closely the real user behaviour (you spot the item you want to delete, then you click on its remove button):

```tsx
await user.click(within(screen.getByTestId('item-1')).getByRole('button', {name: /remove/i}));
```

Doing this also removed the need for devs to add themselves the data-testid on the item.

Reply:
Suppose I have two items in my collection, with the solution I propose I can distinguish between the two remove buttons of each item like this (see snippet) so it satisfies the initial demand.

```tsx
const item1 = screen.getByTestId('item-1');
const item2 = screen.getByTestId('item-2');
const removeItem1Button = within(item1).getByRole('button', {name: /remove/i});
const removeItem2Button = within(item2).getByRole('button', {name: /remove/i});
```